tick/survival/convolutional_sccs.py

```diff
@@ -0,0 +1,922 @@
+            ``step = 1 / model.get_lip_max()``. Otherwise, use a
+            try-an-improve approach.
+
+        tol : `float`, default=1e-5
+            The tolerance of the solver (iterations stop when the stopping
+            criterion is below it).
+
+        max_iter : `int`, default=100
+            Maximum number of iterations of the solver, namely maximum
+            number of epochs.
+
+        verbose : `bool`, default=False
+            If `True`, solver verboses history, otherwise nothing is
+            displayed.
+
+        random_state : `int`, default=None
+            If not None, the seed of the random sampling.
+
+
+        Attributes
+        ----------
+        n_cases : `int` (read-only)
+            Number of samples
+
+        n_intervals : `int` (read-only)
+            Number of time intervals
+
+        n_features : `int` (read-only)
+            Number of features
+
+        n_coeffs : `int` (read-only)
+            Total number of coefficients of the model
+
+        coeffs : `numpy.ndarray`, shape=(n_coeffs,), dtype="float64" (read-only)
+            Coefficients of the model.
+
+        bootstrap_coeffs : `Bootstrap_CI` (read-only)
+            Bootstrap coefficients and confidence intervals of the model.
+        """
+        Base.__init__(self)
+        # Init objects to be computed later
+        self.n_cases = None
+        self.n_intervals = None
+        self.n_features = None
+        self.n_coeffs = None
+        self.coeffs = None
+        self.bootstrap_coeffs = Bootstrap_CI(list(), list(), list(), list(),
+                                             None)
+        self._fitted = None
+        self._step_size = None
+
+        # Init user defined parameters
+        self._features_offset = None
+        self._n_lags = None
+        self.n_lags = n_lags
+        self._penalized_features = None
+        self.penalized_features = penalized_features
+        self._strength_tv = None
+        self._strength_group_l1 = None
+        self.strength_tv = strength_tv
+        self.strength_group_l1 = strength_group_l1
+
+        self._step_size = step
+        self.tol = tol
+        self.max_iter = max_iter
+        self.verbose = verbose,
+        self.print_every = print_every
+        self.record_every = record_every
+        random_state = int(np.random.randint(0, 1000, 1)[0]
+                           if random_state is None else random_state)
+        self._random_state = None
+        self.random_state = random_state
+
+        # Construct objects
+        self._preprocessor_obj = self._construct_preprocessor_obj()
+        self._model_obj = None
+        self._solver_obj = self._construct_solver_obj(step, max_iter, tol,
+                                                      print_every, record_every,
+                                                      verbose, random_state)
+
+    # Interface
+    def fit(self, features: list, labels: list,
+            censoring: np.array, bootstrap: bool = False,
```

Comment:
We should not use "bootstrap" in the names... we should use confidence_interval only, we don't really care that it's done with bootstrap. And we could use something else later...